app/script/conversation/ConversationEphemeralHandler.js

```diff
@@ -0,0 +1,184 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.conversation = z.conversation || {};
+
+z.conversation.ConversationEphemeralHandler = class ConversationEphemeralHandler {
+  static get CONFIG() {
+    return {
+      INTERVAL_TIME: 250,
+    };
+  }
+
+  constructor(conversationService, onMessageTimeout) {
+    this.conversationService = conversationService;
+    this.onMessageTimeout = onMessageTimeout;
+    this.logger = new z.util.Logger('z.conversation.ConversationEphemeralHandler', z.config.LOGGER.OPTIONS);
+
+    this.timedMessages = [];
+    window.setInterval(this._updateTimedMessages.bind(this), ConversationEphemeralHandler.CONFIG.INTERVAL_TIME);
+  }
```

Comment:
This looks good for now to be merged to edge. But let's improve this part and only `setInterval` if `!!timeMessages.length` and clear it when `!timedMessages`. We cann always do that if we turn timed messages into an `ko.obserableArray` and subscribe to it or manually handle it on add/remove/update.